test/loco/integer/magicseries1.clj

```diff
@@ -0,0 +1,57 @@
+(ns ^{:author "Andrea Richiardi"
+      :doc "Magic Series example + tests. It makes use of loco's $cardinality global constraint.
+            A series S = (S0,...Sn) is magic if Si represents the number of occurrences of i in S.
+            For example a Magic Series of length 5 is |2|1|2|0|0|"}
+  loco.integer.magicseries1
+  (:require [clojure.test :refer :all]
+            [loco.core :refer :all]
+            [loco.constraints :refer :all]))
+
+(defn series-vars
+  "Returns the control variables of the Magic Series."
+  [series-length]
+  (for [i (range series-length)]
+    [:s i]))
+
+(defn numeric-constraints
+  "Builds the numeric constraints of each Magic Series control variable
+  in input."
+  [series]
+  (let [series-length (count series)]
+    (for [s series] 
+      ($in s 0 (dec series-length)))))
+
+(defn frequency-map
+  "Builds the frequency map of the series in input for the $cardinality
+  constraint."
+  [series]
+  (into {} (for [[k i] series] [i [k i]])))
+
+(defn ms-model
+  "Return the model."
+  [series-length]
+  (let [series (series-vars series-length)] 
+    (conj (concat series (numeric-constraints series)) 
```

Comment:
`series` by itself is unnecessary here. It won't throw an error within Loco, but it's not doing anything because the names of the variables are just sitting at the top level.